File: java/src/main/java/com/iteso/nintendo/Mewtwo.java
package com.iteso.nintendo;
/**
 * Created by rvillalobos on 2/24/18.
 */
public class Mewtwo extends PokemonCharacter {
    /**
     * Hit Points are the maximum life of pokemon.
     */
    public static final int HIT_POINTS = 105;
    /**
     * Defense multiplier value between 0-1.
     */
    public static final double DEFENSE_MULTIPLIER = 0.6;
    /**
     * Pikachu constructor.
     */
    public Mewtwo() {
        setType("Psychic");
        setName("Mewtwo");
        setHasEvolution(true);
        setSecondaryAttackBehavior(new CosmicPowerBehavior());
        setMainAttackBehavior(new BigFireBehavior());
        setHitPoints(HIT_POINTS);
        setDefenseMultiplier(DEFENSE_MULTIPLIER);
    }

    @Override
    public final String evolve() {
        return null;
    }

    @Override
    public final String defend(final int attack) {
        int damage;

        damage = (int) (attack * getDefenseMultiplier());
        int newHP = getHitPoints() - damage;

        String defendMessage = new String("Defending attack, damage caused is "
                + damage + " new HP is " + newHP);

        setHitPoints(newHP);
        return defendMessage;

    }

    @Override
    public final String secondAttack() {
        return this.getSecondAttackBehavior().attack();
    }

    @Override
    public final String mainAttack() {
        return this.getMainAttackBehavior().attack();
    }

    @Override
    public final void setNewAttack(final int attack,
                                    final AttackBehavior newBeh) {
        if (attack == 1) {
            this.setMainAttackBehavior(newBeh);
        } else {
            this.setSecondaryAttackBehavior(newBeh);
        }
    }
}

File: python/src/main/python/Mewtwo.py
from PokemonCharacter import *
from CosmicPowerBehavior import *
from BigFireBehavior import *


class Charmander(PokemonCharacter):
    def __init__(self):
        self.HIT_POINTS = 105
        self.DEFENSE_MULTIPLIER = 0.6
        self.CosmicPowerBehavior = CosmicPowerBehavior()
        self.BigFireBehavior = BigFireBehavior()
        self.setType("Psychic")
        self.setName("Mewtwo")
        self.setSecondaryAttackBehavior(self.CosmicPowerBehavior)
        self.setMainAttackBehavior(self.BigFireBehavior)
        self.setHasEvolution(True)
        self.setHitPoints(self.HIT_POINTS)
        self.setDefenseMultiplier(self.DEFENSE_MULTIPLIER)

    def evolve(self):
        return None

    def defend(self, attack):
        damage = (attack * self.getDefenseMultiplier())
        newHP = self.getHitPoints() - damage
        self.setHitPoints(newHP)
        string = "Defending attack, damage caused is "
        string += str(int(damage))
        string += " new HP is "
        string += str(int(newHP))
        return string

    def secondAttack(self):
        return self.getSecondAttackBehavior().attack()

    def mainAttack(self):
        return self.getMainAttackBehavior().attack()

    def setNewAttack(self, attack, newAttack):
        if (attack == 1):
            self.setMainAttack(newAttack)
        else:
            self.setSecondAttack(newAttack)